test/compilable/cppmangle.d

```diff
@@ -463,3 +463,32 @@ version (Windows)
 {
     static assert(test15388.mangleof == "?test15388@@YAX$$T@Z");
 }
+
+/**************************************/
+// https://issues.dlang.org/show_bug.cgi?id=18891
+
+extern (C++) class C18891
+{
+    ~this();
+    extern (C++) struct Agg
+    {
+        ~this() {}
```

Comment:
I'm not entirely sure of the DMD AST but won't this be a body in the AST? I'm thinking about this line https://github.com/dlang/dmd/pull/8275/files#diff-12b8cdd12102ad3f468b3a4faa04f0aeR806

Reply:
Oh? I wasn't sure about that line.
I couldn't work out how to identify if a function is a prototype or not. That's what I came up with.

Reply:
Oh? I wasn't sure about that line.
I couldn't work out how to identify if a function is a prototype or not. That's what I came up with.

Reply:
The point of the test is that there should be one destructor without a body and one with?

Reply:
Yes, it should be structured such that the class *wants* to produce an aggregate destructor, but doesn't.

Reply:
The tests confirm that `__dtor` is `__xdtor` (alias), ie, they are the same function; no aggregate destructor was generated.